modal/_container_entrypoint.py

```diff
@@ -554,9 +554,9 @@ def run_inputs(input_id: str, function_call_id: str, args: Any, kwargs: Any) ->
                         item_count += 1
 
                     function_io_manager._queue_put(generator_queue, _FunctionIOManager._GENERATOR_STOP_SENTINEL)
-                    generator_output_task.result()  # Wait to finish sending generator outputs.
                     message = api_pb2.GeneratorDone(items_total=item_count)
                     function_io_manager.push_output(input_id, started_at, message, api_pb2.DATA_FORMAT_GENERATOR_DONE)
+                    generator_output_task.result()  # Wait to finish sending generator outputs.
```

Comment:
Don't know if this change is of much benefit to sync generators.
The equivalent change in the async path is efficacious because it allows the HTTP response to return more quickly to the client.